nesta_ds_utils/networks/generate.py

```diff
@@ -0,0 +1,118 @@
+import numpy as np
+import itertools
+import warnings
+from typing import Union, List
+from collections import Counter, defaultdict
+from itertools import chain, combinations
+import networkx as nx
+import nltk
+from nltk.tokenize import word_tokenize
+
+nltk.download("punkt")
+
+
+def term_cooccurrence_graph(
+    sequences: Union[List[list], List[np.array], List[str]],
+    graph_type: str = "networkx",
+    weighted: bool = True,
+    directed: bool = False,
+    stopwords: list = [],
+    lowercase: bool = True,
+) -> nx.Graph:
+    """generates a co-occurence graph based on pairwise co-occurence of terms.
+
+    Args:
+        sequences (Union[List[list], List[np.array], List[str]]):
+            list of term lists, list of numpy arrays containing terms, or list of strings.
+        graph_type (str, optional): Python library to use for network generation.
+            Currently only supports 'networkx' but future development should support 'graph-tool'
+        weighted (bool, optional): parameter to indicate of edges should be weighted. Defaults to True.
+            If True, edge weights represent number of co-occurences.
+        directed (bool, optional): parameter to indicate if edges should be directed. Defaults to False.
+            If True, edge directions are determined based on order of tokens. CURRENTLY NOT SUPPORTED.
+        stopwords (list, optional): list of stopwords tokens to exclude from vertices. Defaults to [].
+        lowercase (bool, optional): parameter to indicate if tokens should be lowercase. Defaults to True.
+            If True, all tokens are lowercased.
+            If False, tokens are treated as different nodes based on capitalization.
+
+    Returns:
+        nx.Graph: Returns networkx graph object. If directed=True returns nx.DiGraph, otherwise returns nx.Graph.
+        Currently only supports undirected.
+    """
+    if directed == True:
+        warnings.warn(
+            "Directed graphs are not currently supported by this function. Generating undirected Graph."
+        )
```

Comment:
I think this would be pretty trivial to implement. If True then you just instantiate a DiGraph and add the same edges in the opposite order. The directed graph is only useful once you start to do assymmetric edge weight normalisation, but still feel like it's a small addition to make.

Reply:
What would be the added value of returning a DGraph with symmetric weights?